common/pkg/libartifact/store/store.go

```diff
@@ -297,9 +292,8 @@ func (as ArtifactStore) Add(ctx context.Context, dest string, artifactBlobs []li
 	}
 	defer imageDest.Close()
 
-	// Unlock around the actual pull of the blobs.
-	// This is ugly as hell, but should be safe.
-	locked = false
+	// Release the lock during blob copying to allow concurrent operations.
+	// We'll reacquire it later and check for conflicts before committing.
 	as.lock.Unlock()
```

Comment:
If you're keeping this you absolutely need to keep the `locked` variable, this risks a double-unlock if the defer fires before the lock reacquires